### ISSUE TITLE
Assets file is missing compile assets when generated within Visual Studio vs command-line causing build errors

### ISSUE THREAD

**Issue by davkean** [NuGet/Home#7446]
## Details about Problem

Visual Studio: Version 16.0.0 Preview 1.0 [28227.50.d16.0]
Dotnet: 3.0.100-alpha1-009667

## Detailed repro steps so we can see the same problem

Given this project:
``` XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net35</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.VisualStudio.Setup.Configuration.Interop" Version="1.16.30" />
  </ItemGroup>

</Project>
```

1. From command-line `dotnet restore ConsoleApp.csproj`
2. Rename project.assets.json to project.assets.json.commandline
3. Open project in VS and let it restore
4. Compare the two assets file

Command-line:

``` JSON
{
  "version": 3,
  "targets": {
    ".NETFramework,Version=v3.5": {
      "Microsoft.VisualStudio.Setup.Configuration.Interop/1.16.30": {
        "type": "package",
        "compile": {
          "lib/net35/Microsoft.VisualStudio.Setup.Configuration.Interop.dll": {}
        },
        "runtime": {
          "lib/net35/Microsoft.VisualStudio.Setup.Configuration.Interop.dll": {}
        }
      }
    },
    ".NETFramework,Version=v3.5/win7-x86": {
      "Microsoft.VisualStudio.Setup.Configuration.Interop/1.16.30": {
        "type": "package",
        "compile": {
          "lib/net35/Microsoft.VisualStudio.Setup.Configuration.Interop.dll": {}
        },
        "runtime": {
          "lib/net35/Microsoft.VisualStudio.Setup.Configuration.Interop.dll": {}
        }
      }
    }
  },
```

Visual Studio (notice `compile` is missing reference to dll):

``` JSON
{
  "version": 3,
  "targets": {
    ".NETFramework,Version=v3.5": {
      "Microsoft.VisualStudio.Setup.Configuration.Interop/1.16.30": {
        "type": "package",
        "compile": {
          "lib/net35/_._": {}
        },
        "runtime": {
          "lib/net35/Microsoft.VisualStudio.Setup.Configuration.Interop.dll": {}
        }
      }
    },
    ".NETFramework,Version=v3.5/win7-x86": {
      "Microsoft.VisualStudio.Setup.Configuration.Interop/1.16.30": {
        "type": "package",
        "compile": {
          "lib/net35/_._": {}
        },
        "runtime": {
          "lib/net35/Microsoft.VisualStudio.Setup.Configuration.Interop.dll": {}
        }
      }
    }
  },
```

This is blocking building http://github.com/dotnet/project-system inside Visual Studio.

**Comment by jainaashish** [comment]
This happens because of PR# https://github.com/NuGet/NuGet.Client/pull/2466 which tried to fix it for packages with `developmentDependency` but regressed it for already installed packages as well.

Correct fix is in PR# https://github.com/NuGet/NuGet.Client/pull/2494

**Comment by jainaashish** [comment]
PR# 2494 merged into 4.9.0-preview5 as well as dev branch
